attrd/commands.c

```diff
@@ -671,14 +671,53 @@ attrd_peer_update(crm_node_t *peer, xmlNode *xml, const char *host, bool filter)
 {
     bool changed = FALSE;
     attribute_value_t *v = NULL;
+    int dampen = 0;
 
+    const char *op = crm_element_value(xml, F_ATTRD_TASK);
     const char *attr = crm_element_value(xml, F_ATTRD_ATTRIBUTE);
     const char *value = crm_element_value(xml, F_ATTRD_VALUE);
+    const char *dvalue = crm_element_value(xml, F_ATTRD_DAMPEN);
 
     attribute_t *a = g_hash_table_lookup(attributes, attr);
 
     if(a == NULL) {
-        a = create_attribute(xml);
+        if (safe_str_eq(op, ATTRD_OP_UPDATE) || safe_str_eq(op, ATTRD_OP_UPDATE_BOTH)) {
+            a = create_attribute(xml);
+        } else {
+            crm_warn("Update error (attribute %s not found)", attr);
+            return;
+        }
+    } else {
```

Comment:
Hi Hideo,

I just realized this 'else' should be removed. If 'a' is originally NULL and 'op' is ATTRD_OP_UPDATE_BOTH, we still need to change the dampen. We've already returned if 'a' is NULL and 'op' is ATTRD_OP_UPDATE_DELAY, so there should be no problem with simply removing the 'else' and keeping the next 'if'.